onnxruntime/core/providers/cuda/cuda_kernel.h

```diff
@@ -48,8 +48,8 @@ class CudaKernel : public OpKernel {
   }
 
   template <typename T>
-  inline IAllocatorUniquePtr<T> GetScratchBuffer(size_t count_or_bytes) const {
-    return provider_->GetScratchBuffer<T>(count_or_bytes);
+  inline IAllocatorUniquePtr<T> GetScratchBuffer(size_t count_or_bytes, bool is_reserve = false) const {
```

Comment:
The name of "is_reserve" flag is counterintuitive. Allocated buffer will be kept in Arena if "is_reserve==false", while it will not be reserved by Arena if "is_reserved==true". 

I suggest we create a new API for this purpose, since GetScratchBuffer() is very widely used and let's not add cognitive burden for developer when using this. 

 We can have another API, e.g. GetTransientScratchBuffer, and please document its behavior.